src/private/CBCRuntime.m

```diff
@@ -15,58 +15,120 @@
  */
 
 #import "CBCRuntime.h"
-
 #import "CBCCatalogExample.h"
-
 #import <objc/runtime.h>
 
+NSArray<NSString *> *CBCCatalogBreadcrumbsFromClass(Class aClass);
+BOOL CBCCatalogIsPrimaryDemoFromClass(Class aClass);
+BOOL CBCCatalogIsPresentableFromClass(Class aClass);
+BOOL CBCCatalogIsDebugLeaf(Class aClass);
+NSString *CBCDescriptionFromClass(Class aClass);
+NSURL *CBCRelatedInfoFromClass(Class aClass);
+NSDictionary *CBCConstructMetadataFromMethods(Class aClass);
+NSString *CBCStoryboardNameFromClass(Class aClass);
+
```

Comment:
Let's revert this change, it shouldn't be necessary. You can do this kind of thing to forward-declare functions in headers, but it's unusual in implementations.

The compiler is telling you that you're implementing a function for exporting to other files that hadn't been declared before, which is weird.

Are you still using these functions (anywhere)? If not, delete their implementations here in `CBCRuntime.m`. If yes, then make sure they're declared in the `CBCRuntime.h`, as they were before.

Reply:
We are still using them to construct the metadata dictionary using the methods in the examples for backwards compatibility.

Because these are C functions, in C99 I must declare their prototype. I can move these declarations to the header file, but they are no longer used outside the scope of that file, and no longer needed to be externed.

Reply:
I believe this is a result of Bazel's strict C symbol parsing. I've seen the same thing with other C functions in MDC code. It's requiring an explicit declaration before the definition of the function.

Reply:
We could either move these declarations back to the header file (without an extern), or we can stop backwards compatibility and make a breaking change, but then our downstream apps have to move to the new way of implementing `catalogMetadata` rather than still be able to use the current CbC methods (`catalogBreadcrumbs`, `catalogRelatedInfo` etc.)

Reply:
The last option is to stop using C entirely for global methods, and have these set up as obj-c class methods instead, they will then be invoked as `[CBCRuntime X];` and do the same for the C methods for the `CBCNode` class

Reply:
No, this is fine and C is appropriate here.

If these are only used in this file, then that's fine, just make them scoped locally to this file. Move them to the top after the imports and mark them as `static`, e.g.

    #import <SomeFile.h>

    static BOOL myFunc1(void) {
      // ...
    }

    static BOOL myFunc2(void) {
      // ...
    }

    // External functions/methods that use myFunc1 and myFunc2.
    // ...

That way they can only be used in this file and the compiler will be happy. :)

Reply:
sounds like a plan, thanks!